src/applications/personalization/profile/components/paperless-delivery/PaperlessDelivery.jsx

```diff
@@ -1,27 +1,77 @@
-import React from 'react';
-import { useSelector } from 'react-redux';
+import React, { useEffect, useMemo } from 'react';
+import { shallowEqual, useDispatch, useSelector } from 'react-redux';
 import { PROFILE_PATH_NAMES } from '@@profile/constants';
-import { selectVAPEmailAddress } from '~/platform/user/selectors';
+import { fetchCommunicationPreferenceGroups } from '@@profile/ducks/communicationPreferences';
+import {
+  hasVAPServiceConnectionError,
+  selectVAPEmailAddress,
+} from '~/platform/user/selectors';
+import { selectPatientFacilities } from '~/platform/user/cerner-dsot/selectors';
+import { VaLoadingIndicator } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+import { selectCommunicationPreferences } from '@@profile/reducers';
 import Headline from '../ProfileSectionHeadline';
+import { FieldHasBeenUpdated } from '../alerts/FieldHasBeenUpdated';
+import { LOADING_STATES } from '../../../common/constants';
 import { Description } from './Description';
 import { Note } from './Note';
 import { MissingEmailAlert } from './MissingEmailAlert';
 import { ProfileEmail } from './ProfileEmail';
-import { FieldHasBeenUpdated } from '../alerts/FieldHasBeenUpdated';
+import { Documents } from './Documents';
 
 export const PaperlessDelivery = () => {
+  const dispatch = useDispatch();
   const emailAddress = useSelector(selectVAPEmailAddress);
-  const showMissingEmailAlert = !emailAddress;
+  const facilities = useSelector(selectPatientFacilities, shallowEqual);
+  const communicationPreferencesState = useSelector(
+    selectCommunicationPreferences,
+  );
+  const hasVAPServiceError = useSelector(hasVAPServiceConnectionError);
+  const hasLoadingError = Boolean(communicationPreferencesState.loadingErrors);
+  const shouldShowAPIError = hasVAPServiceError || hasLoadingError;
+  const shouldShowLoadingIndicator =
+    communicationPreferencesState.loadingStatus === LOADING_STATES.idle ||
+    communicationPreferencesState.loadingStatus === LOADING_STATES.pending;
+  const shouldFetchNotificationSettings = !shouldShowAPIError;
+  const shouldShowNotificationGroups = useMemo(
+    () => {
+      return !shouldShowAPIError && !shouldShowLoadingIndicator;
+    },
+    [shouldShowAPIError, shouldShowLoadingIndicator],
+  );
+
+  useEffect(
+    () => {
+      if (shouldFetchNotificationSettings) {
+        dispatch(
+          fetchCommunicationPreferenceGroups({
+            facilities,
```

Comment:
I was looking at some communication preferences code recently and was really confused by passing in the facilities into fetchCommunicationPreferenceGroups like this because the communication preferences endpoint doesn't accept facilities. I still don't really understand why we're shoving facilities into redux for communication preferences.

Reply:
It looks like they're only using `facilities` to help with their filtering logic in the communication preferences state.